link_linux.go

```diff
@@ -2136,6 +2195,18 @@ func LinkDeserialize(hdr *unix.NlMsghdr, m []byte) (Link, error) {
 				protinfo := parseProtinfo(attrs)
 				base.Protinfo = &protinfo
 			}
+		case unix.IFLA_PROP_LIST | unix.NLA_F_NESTED:
+			attrs, err := nl.ParseRouteAttr(attr.Value[:])
+			if err != nil {
+				return nil, err
+			}
+
+			base.AltNames = []string{}
+			for _, attr := range attrs {
+				if attr.Attr.Type == unix.IFLA_ALT_IFNAME {
+					base.AltNames = append(base.AltNames, string(attr.Value[:len(attr.Value)-1]))
```

Comment:
you can use: https://github.com/vishvananda/netlink/blob/857968af1192b419c958984e96fa5154ada6baff/nl/nl_linux.go#L869

which takes care of the `0` char

but i see most of the code here does as you do.